ass_man/models.py

```diff
@@ -20,62 +21,84 @@ def __str__(self):
         return (self.vendor + ' ' + self.model_number) or ''
 
 
-class Instance(models.Model):
+class Asset(models.Model):
     model = models.ForeignKey(Model, on_delete=models.PROTECT)
-    hostname = models.CharField(max_length=64)
+    hostname = models.CharField(max_length=64, blank=True, null=True)
+    datacenter = models.ForeignKey('Datacenter', on_delete=models.CASCADE, null=True)
     rack = models.ForeignKey('Rack', on_delete=models.PROTECT)
     rack_u = models.PositiveIntegerField(blank=False)
     owner = models.ForeignKey(User, blank=True, null=True, on_delete=models.PROTECT)
     comment = models.TextField(blank=True)
+    asset_number = models.PositiveIntegerField(blank=True, default=100000)
 
     def __str__(self):
         return self.hostname or ''
 
+class Network_Port(models.Model):
+    name = models.CharField(max_length=15, blank=True, default='mgmt')
+    mac = models.CharField(max_length=17, blank=True, null=True)
+    connection = models.ForeignKey('self', on_delete=models.SET_NULL, null=True)
+    asset = models.ForeignKey('Asset', on_delete=models.CASCADE)
+
+class Power_Port(models.Model):
+    pdu = models.ForeignKey('PDU', on_delete=models.CASCADE)
```

Comment:
we don't want to delete pdu when hte PP is deleted (or vice versa right), maybe this should be set null?

Reply:
I was operating like Power Port was a part of PDU and connected to an Asset. Not part of an Asset and connected to a PDU. So that's why I had these flipped. But yeah I'll change it